test/e2e/toolchaincluster_test.go

```diff
@@ -86,19 +90,45 @@ func verifyToolchainCluster(t *testing.T, await *wait.Awaitility, otherAwait *wa
 		err := await.Client.Create(context.TODO(), toolchainCluster)
 		// then the ToolchainCluster should be offline
 		require.NoError(t, err)
-		_, err = await.WaitForNamedToolchainClusterWithCondition(toolchainCluster.Name, &toolchainv1alpha1.ToolchainClusterCondition{
+		namedToolchainCluster, err := await.WaitForNamedToolchainClusterWithCondition(toolchainCluster.Name, &toolchainv1alpha1.ToolchainClusterCondition{
 			Type:   toolchainv1alpha1.ToolchainClusterOffline,
 			Status: corev1.ConditionTrue,
 		})
 		require.NoError(t, err)
+		checkTenantRoleLabel(t, &namedToolchainCluster, await)
 		// other ToolchainCluster should be ready, too
-		_, err = await.WaitForToolchainClusterWithCondition(otherAwait.Type, otherAwait.Namespace, wait.ReadyToolchainCluster)
+		toolChainClusterWithCondition, err := await.WaitForToolchainClusterWithCondition(otherAwait.Type, otherAwait.Namespace, wait.ReadyToolchainCluster)
 		require.NoError(t, err)
-		_, err = otherAwait.WaitForToolchainClusterWithCondition(await.Type, await.Namespace, wait.ReadyToolchainCluster)
+		checkTenantRoleLabel(t, &toolChainClusterWithCondition, await)
+		otherToolchainCluster, err := otherAwait.WaitForToolchainClusterWithCondition(await.Type, await.Namespace, wait.ReadyToolchainCluster)
 		require.NoError(t, err)
+		checkTenantRoleLabel(t, &otherToolchainCluster, otherAwait)
 	})
 }
 
+// checkTenantRoleLabel checks that cluster-role label `tenant` is set accordingly since not all cluster types
+// should have this label (only member ones).
+// In case the label is expected but missing at the first check,
+// the function waits for the label to be added by the toolchaincluster controller.
+func checkTenantRoleLabel(t *testing.T, toolchainCluster *toolchainv1alpha1.ToolchainCluster, await *wait.Awaitility) {
+	clusterType, clusterTypeLabelExists := toolchainCluster.Labels[cluster.LabelType]
+	if clusterTypeLabelExists {
```

Comment:
I think that they have to contain the type label - so, you could also assert that. Or it's maybe already verified before, so no need for the if-statement.